core/toolbox.js

```diff
@@ -164,18 +230,31 @@ Blockly.Toolbox.prototype.position = function() {
   var svg = this.workspace_.getParentSvg();
   var svgPosition = goog.style.getPageOffset(svg);
   var svgSize = Blockly.svgSize(svg);
-  if (this.workspace_.RTL) {
-    treeDiv.style.left =
-        (svgPosition.x + svgSize.width - treeDiv.offsetWidth) + 'px';
-  } else {
+  if (this.horizontalLayout_) {
     treeDiv.style.left = svgPosition.x + 'px';
-  }
-  treeDiv.style.height = svgSize.height + 'px';
-  treeDiv.style.top = svgPosition.y + 'px';
-  this.width = treeDiv.offsetWidth;
-  if (!this.workspace_.RTL) {
-    // For some reason the LTR toolbox now reports as 1px too wide.
-    this.width -= 1;
+    treeDiv.style.height = 'auto';
+    treeDiv.style.width = svgSize.width + 'px';
+    this.height = treeDiv.offsetHeight;
+    if (this.toolboxPosition == Blockly.TOOLBOX_AT_TOP) {  // Top
+      treeDiv.style.top = svgPosition.y + 'px';
+    } else {  // Bottom
+      var topOfToolbox = svgPosition.y + svgSize.height - treeDiv.offsetHeight;
+      treeDiv.style.top = topOfToolbox + 'px';
+    }
+  } else {
+    if (this.toolboxPosition == Blockly.TOOLBOX_AT_RIGHT) {  // Right
+      treeDiv.style.left =
+          (svgPosition.x + svgSize.width - treeDiv.offsetWidth) + 'px';
+    } else { // Left
```

Comment:
Add a second leading space before the comment.  To match the other three comments above.
(This is a style rule for Python at Google, we don't have a rule one way or another for JS.  But the codebase is consistent with a double space before an inline comment.)